include/boost/url/detail/impl/remove_dot_segments.ipp

```diff
@@ -0,0 +1,518 @@
+            // prefix of "/./" or "/.", where "." is
+            // a complete path segment, then replace
+            // that prefix with "/" in the input
+            // buffer; otherwise,
+            n = detail::pct_decode_ends_with_unchecked(s, "/./");
+            if (n)
+            {
+                c = s.substr(s.size() - n);
+                s.remove_suffix(n);
+                continue;
+            }
+            n = detail::pct_decode_ends_with_unchecked(s, "/.");
+            if (n)
+            {
+                c = s.substr(s.size() - n);
+                s.remove_suffix(n);
+                continue;
+            }
+
+            // C. if the input buffer begins with a
+            // prefix of "/../" or "/..", where ".."
+            // is a complete path segment, then
+            // replace that prefix with "/" in the
+            // input buffer and remove the last
+            // segment and its preceding "/"
+            // (if any) from the output buffer
+            // otherwise,
+            n = detail::pct_decode_ends_with_unchecked(s, "/../");
+            if (n)
+            {
+                c = s.substr(s.size() - n);
+                s.remove_suffix(n);
+                ++l;
+                continue;
+            }
+            n = detail::pct_decode_ends_with_unchecked(s, "/..");
+            if (n)
+            {
+                c = s.substr(s.size() - n);
+                s.remove_suffix(n);
+                ++l;
+                continue;
+            }
+
+            // E.  move the first path segment in the
+            // input buffer to the end of the output
+            // buffer, including the initial "/"
+            // character (if any) and any subsequent
+            // characters up to, but not including,
+            // the next "/" character or the end of
+            // the input buffer.
+            std::size_t p = s.size() > 1 ?
+                s.find_last_of('/', s.size() - 2)
+                : string_view::npos;
+            if (p != string_view::npos)
+            {
+                c = s.substr(p + 1);
+                s.remove_suffix(c.size());
+            }
+            else
+            {
+                c = s;
+                s = {};
+            }
+
+            if (l == 0)
+            {
+                return;
+            }
+            if (!s.empty())
+                --l;
+        }
+        // we still need to skip n_skip + 1
+        // but the string is empty
+        if (r && l)
+        {
+            c = "/";
+            l = 0;
+            return;
+        }
+        else if (l)
+        {
+            if (c.empty())
+            {
+                c = "/..";
+            }
+            else
+            {
+                c = "/../";
+            }
+            --l;
+            return;
+        }
+        c = {};
+    };
+
+    // Calculate the normalized size
+    auto norm_bytes = [&pop_last](
+         string_view p,
+         bool r)
```

Comment:
Do you have tabs set to 5 spaces or something?

Reply:
I'm not using automatic formatting with boost.url. I'm mostly trying to mimic the unspoken rules.